pontoon/sync/utils.py

```diff
@@ -1,184 +1,109 @@
-    """
-    return os.path.abspath(os.path.join(path, os.pardir))
-
-
-def uses_undercore_as_separator(directory: str) -> bool:
-    """
-    Return True if the names of folders in a directory contain more '_' than '-'.
-    """
-    only_folders = []
-    subdirs = os.listdir(directory)
-
-    for i in subdirs:
-        if os.path.isdir(os.path.join(directory, i)):
-            only_folders.append(i)
-
-    return "".join(only_folders).count("_") > "".join(only_folders).count("-")
-
-
-def directory_contains_resources(directory_path: str, source_only=False) -> bool:
-    """
-    Return True if the given directory contains at least one
-    supported resource file (checked via file extension), or False
-    otherwise.
-
-    :param source_only:
-        If True, only check for source-only formats.
-    """
-    resource_check = is_source_resource if source_only else is_resource
-    for root, dirnames, filenames in os.walk(directory_path):
-        # first() avoids checking past the first matching resource.
-        if first(filenames, resource_check) is not None:
-            return True
-    return False
-
-
-def locale_directory_path(
-    checkout_path: str, locale_code: str, parent_directories: list[str]
-) -> str:
-    """
-    Path to the directory where strings for the given locale are
-    stored.
-    """
-
-    # Check paths that use underscore as locale/country code separator
-    locale_code_variants = [locale_code, locale_code.replace("-", "_")]
-
-    # Optimization for directories with a lot of paths: if parent_directories
-    # is provided, we simply join it with locale_code and check if path exists
-    possible_paths = [
-        path
-        for path in (
-            os.path.join(parent_directory, locale)
-            for locale in locale_code_variants
-            for parent_directory in parent_directories
+import re
+
+from io import BytesIO
+from os.path import basename, exists, join, relpath
+from tempfile import TemporaryDirectory
+from zipfile import ZipFile
+
+from django.core.files import File
+from django.utils import timezone
+
+from pontoon.base.models import ChangedEntityLocale, Locale, Project, User
+from pontoon.base.models.repository import Repository
+from pontoon.sync.core.checkout import checkout_repos
+from pontoon.sync.core.paths import UploadPaths, find_paths
+from pontoon.sync.core.stats import update_stats
+from pontoon.sync.core.translations_from_repo import find_db_updates, write_db_updates
+from pontoon.sync.core.translations_to_repo import update_changed_resources
+
+
+# FIXME This is a temporary hack, to be replaced by 04/2025 with proper downloads.
+def translations_target_url(
+    project: Project, locale: Locale, resource_path: str
+) -> str | None:
+    """The target repository URL for a resource, for direct download."""
+
+    if project.repositories.count() > 1:
+        # HACK: Let's assume that no config is used, and the target repo root is the right base.
```

Comment:
With this hack we probably don't even need to redirect users to GitHub (given we still have the ability to use shallow clones)? 

Probably not worth making a change now. 

We should however file a bug to improve performance of `find_paths()`.

Reply:
Done: mozilla/moz-l10n#37